iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/chunk_distributor_data.hpp

```diff
@@ -31,16 +32,23 @@ namespace popo
 {
 class ThreadSafePolicy
 {
-  public: // needs to be public since we want to use std::lock_guard
+  public:
+    // needs to be public since we want to use std::lock_guard
     void lock() const
     {
-        m_mutex.lock();
+        if (!m_mutex.lock())
+        {
+            errorHandler(Error::kPOPO__CHUNK_DISTRIBUTOR_LOCKING_ERROR, nullptr, ErrorLevel::FATAL);
+        }
     }
-    void unlock() const 
+    void unlock() const
     {
-        m_mutex.unlock();
+        if (!m_mutex.unlock())
+        {
+            errorHandler(Error::kPOPO__CHUNK_DISTRIBUTOR_LOCKING_ERROR, nullptr, ErrorLevel::FATAL);
+        }
     }
-    bool tryLock() const 
+    bool tryLock() const
     {
         return m_mutex.try_lock();
```

Comment:
My assumption is that it returns false when the mutex is already locked. Which is a valid behavior. Do we have to distinguish an error like it can happen with lock/unlock and the "busy" error which is OK for a try lock?

Reply:
According to the documentation https://linux.die.net/man/3/pthread_mutex_trylock  the underlying call `pthread_mutex_trylock` can fail when the thread already owns the mutex which would lead to undefined behavior or when the handle to the mutex is invalid. This should in theory never happen since we have our wrapper which ensures the lifetime but when it does the returning bool would not be enough... And in the other calls like `lock` and `unlock` we return false when an failure like invalid handle happens.